### DISCUSSION TITLE
Automatically update property value if another one changes?

### DISCUSSION THREAD

**Discussion by ftzi** [pmndrs/zustand#1286] [Q&A]
I have a property A in my state that's quite big. I want to have another state B that is this A transformed.

I could have an useMemo in a component, but I don't want a single component to be responsible for doing this.

In Context, I would use useMemo. In MobX, I would use [computeds](https://mobx.js.org/computeds.html).

Is there some way in Zustand to do it?

**Comment by dai-shi** [comment]
Does #1221 help?

**Reply by ftzi** [reply]
It does, but isn't really a solution I would use, as it would be another export / hook to use.

Also, if it's used by more than one component, it seems that the complex value would be calculated for each component using that hook, not a good solution for my case.

I would like to keep everything inside the same store.

This is one important feature for me missing from that library that exists in other alternatives, as I mentioned.

**Reply by dai-shi** [reply]
Does this work for you?

### Use 3rd-party libraries

For example: https://github.com/cmlarsen/zustand-middleware-computed-state

---

> it seems that the complex value would be calculated for each component using that hook

In such a case, you can consider memoizing your selector functions.
